src/Compilers/VisualBasic/Portable/Binding/Binder_Invocation.vb

```diff
@@ -2019,15 +2019,34 @@ ProduceBoundNode:
                 Dim paramIndex = 0
                 Dim someArgumentsBad As Boolean = False
                 Dim someParamArrayArgumentsBad As Boolean = False
+                Dim seenOutOfPositionNamedArgIndex As Integer = -1
 
                 Dim candidateSymbol As Symbol = candidate.UnderlyingSymbol
                 Dim candidateIsExtension As Boolean = candidate.IsExtensionMethod
 
                 For i As Integer = 0 To arguments.Length - 1 Step 1
 
+                    ' A named argument which is used in-position counts as positional
                     If Not argumentNames.IsDefault AndAlso argumentNames(i) IsNot Nothing Then
-                        ' First named argument
-                        Exit For
+                        If Not candidate.TryGetNamedParamIndex(argumentNames(i), paramIndex) Then
```

Comment:
Note: I simplified the error handling here, relying on the checks in the loop for named arguments. That avoids duplicate errors. Also, I think it makes sense: a name that is not found is not "in-position", so we cannot assume it is positional.